docs/development/design/new-sphinx-guides.rst

```diff
@@ -0,0 +1,186 @@
+
+Tutorial
+--------
+
+   **Objective**: Write a tutorial, some examples, and reference
+   documentation for our Python code.
+
+Every step should lead to a “yay!” moment, which marks the end of the
+section. This keeps the learner motivated.
+
+Appendixes are optional, i.e. not required to follow the tutorial, but
+highly recommended.
+
+1. Getting started
+
+   1. Creating our code project
+
+      -  What is our library (present some minimal ``x.py`` file in a
+         dedicated directory, will be the basis of the tutorial)
+      -  Install Python
+      -  Check that our code works (yay!)
+
+   2. Using virtual environments for development
+
+      -  Create a virtual environment (and/or conda environment)
+      -  Activate our virtual environment (it will always be the first
+         step)
+      -  Check that our code also works (yay!)
+
+   3. Adding external dependencies
+
+      -  Add import from small third party library
+      -  Install it inside our virtual environment
+      -  Check that our code works (yay!) and it doesn’t outside the
+         virtual environment (as expected)
+
+   4. Appendix: Using version control
+
+      -  Install git (we will use it during the tutorial)
+      -  Write a barebones ``README.md`` (we will expand it later), a
+         ``.gitignore`` file (copied from gitignore.io), and a proper
+         ``LICENSE`` file
+
+         -  ``LICENSE.md``? ``LICENSE``? ``COPYING``? Whatever
+
+      -  Create the first commit for the project (yay!)
+
+2. First steps to document our project using Sphinx
+
+   1. Installing Sphinx
+
+      -  Activate our virtual environment
+      -  Install Sphinx inside the virtual environment
+      -  Check that ``sphinx-build --help`` works (yay!)
+
+   2. Creating the project layout
+
+      -  Run the sphinx-quickstart wizard: separate source and build
+         (``y``), project name, author name, and go
+      -  Check that the correct files are created (yay!)
+
+   3. Converting our documentation to local HTML
+
+      -  Build the HTML output using
+         ``sphinx-build -b -W html doc doc/_build/html``  [1]_
+      -  Navigate to ``doc/_build/html`` and launch an HTTP server
+         (``python -m http.server``)
+      -  Open http://localhost:8000 in a web browser, and see the HTML
+         documentation (yay!)
+
+   4. Making a change and regenerating the documentation
+
+      -  Add a sentence to ``index.rst`` (no need to explain in detail
+         what reST is at this stage)
+      -  Rebuild HTML ``sphinx-build -W -b html ...``
+      -  Observe that the HTML docs have changed (yay!) (we will have to
+         do this every time we change something)
+
+   5. Simplify documentation building by using Make
+
+      -  Install Make (and its Windows counterpart)
+      -  Make another trivial change
+      -  Build HTML doing ``cd doc && make html``
+      -  Observe that the HTML docs have changed (yay!)
+
+   6. Converting our documentation to PDF
```

Comment:
This seems almost impossible to do well. I don't think we should include this in the tutorial. I almost never created PDFs locally, and we shouldn't expect users to manage this kind of task.

Reply:
You think? I still did not explore how complicated it is to install TeX Live on Windows and macOS, but on Linux it was fairly straightforward. It required a number of dependencies, that's true.